acceptance/testdata/config-noreplace-old.yaml

```diff
@@ -0,0 +1,13 @@
+name: "foo"
+arch: "all"
+platform: "linux"
+version: "v1.0.0"
+maintainer: "Foo Bar"
+vendor: "foobar"
+homepage: "https://foobar.org"
+license: "MIT"
+config_files:
+  ../testdata/whatever.conf: "/etc/regular.conf"
```

Comment:
I was thinking, maybe we could change this thing to be a `map[string]struct`, and then make it something like:

```yaml
config_files:
   ../testdata/whatever.conf:
     path: "/etc/regular.conf"
     replace: true
```

wdyt?

Reply:
although, not sure this can be changed in debs 🤔

Reply:
According to https://www.debian.org/doc/debian-policy/ap-pkg-conffiles.html (5.3, paragraph 4) debs handle this by interrupting the upgrade to query the user about what should happen with the configuration file.

While your approach is quite elegant if both package types had the same concept, I think it is not really suitable for the following reasons:
* It breaks existing configurations
* Future package types could handle it differently, which is confusing for the user when config options may or may not have an effect depending on the package type
* We should not make the configuration more complicated to accommodate for an edge case. I think most people are fine with the regular `config_files` so only those who have a particular use case and know what they are doing can opt for different behaviour by using the separate `config_noreplace_files` option